src/analysis/sign-lattice.cpp

```diff
@@ -0,0 +1,70 @@
+#include "lattice.h"
+#include <type_traits>
+
+namespace wasm::analysis {
+
+struct SignLattice {
+public:
+  enum Sign { BOTTOM, NEGATIVE, ZERO, POSITIVE, TOP };
+
+private:
+  Sign value;
+
+public:
+  static bool isTop(const SignLattice& element) { return element.value == TOP; }
+
+  static bool isBottom(const SignLattice& element) {
+    return element.value == BOTTOM;
+  }
+
+  static LatticeComparison compare(const SignLattice& left,
+                                   const SignLattice& right) {
+    if (left.value == right.value) {
+      return EQUAL;
+    } else if (left.value == BOTTOM || right.value == TOP) {
+      return LESS;
+    } else if (left.value == TOP || right.value == BOTTOM) {
+      return GREATER;
+    } else {
+      return NO_RELATION;
+    }
+  }
```

Comment:
Might consider it for this case, but as we mentioned in our discussion, as a general rule, calculating the LUB is going to be very slow for certain lattices, so we want to avoid using it.